PacMan/src/gameThreads/Game.java

```diff
@@ -180,13 +194,57 @@ private void update() {
 		
 	}
 	
+	public void NextLevel() {
+		List<Entity> entities = entityManager.getAllEntitiesPosessingComponentOfClass(GraphicsComponent.class.getName());
+		board.refreshlevel(level);
+		for(Entity entity: entities) {
+			PhysicsComponent physic = (PhysicsComponent) entityManager.getComponentOfClass(PhysicsComponent.class.getName(), entity);
+			GraphicsComponent graphic = (GraphicsComponent) entityManager.getComponentOfClass(GraphicsComponent.class.getName(), entity);
+			MoveComponent move = (MoveComponent) entityManager.getComponentOfClass(MoveComponent.class.getName(), entity);
+			AudioComponent audio = (AudioComponent) entityManager.getComponentOfClass(AudioComponent.class.getName(), entity);
+			ScoreComponent scores= (ScoreComponent) entityManager.getComponentOfClass(ScoreComponent.class.getName(), entity);
+			LifeComponent life=(LifeComponent) entityManager.getComponentOfClass(LifeComponent.class.getName(), entity);
+			UserInputComponent user=(UserInputComponent) entityManager.getComponentOfClass(UserInputComponent.class.getName(), entity);
+			if(graphic != null && physic!=null && move==null && audio==null &&scores==null && life==null&& user==null) {
+				return ;
+		    
+		    } 
```

Comment:
Fait attention à ton indentation, c'est souvent pas bien espacé

Reply:
ok

Reply:
Ta méthode est supposé faire quoi ? As-tu testé cela ? Tu fait un return dans un For loop...... tu veux un continue pcq dans l'état actuel du code tu vas early exit dès que tu vas arriver sur une gum. Aussi ton indentation et ton espacement n'est pas constant. Tu veux avoir des espaces autour de tes opérateurs comme != , == , etc.